lib/impure/db_sqlite.nim

```diff
@@ -716,8 +719,64 @@ proc setEncoding*(connection: DbConn, encoding: string): bool {.
 proc finalize*(sqlPrepared:SqlPrepared){.discardable.} = 
   discard finalize(sqlPrepared.PStmt)
 
+proc bindInt32*( ps : SqlPrepared, paramIdx: int,
+  val: int32 ):int = 
+  ## Binds a int32 float to the specified paramIndex.
+  result = bind_int( ps.PStmt, paramIdx.int32, val )
+
+proc bindInt64*( ps: SqlPrepared, paramIdx: int,
+    val: int64 ):int =   
+  result = bind_int64(ps.PStmt, paramIdx.int32, val)
+
+proc bindFloat64*( ps: SqlPrepared, paramIdx: int,
+                   val: float64) : int {.inline.} =
+  ## Binds a 64bit float to the specified paramIndex.
+  result = bind_double(ps.PStmt, paramIdx.int32, val)
+
+proc bindNull*( ps : SqlPrepared, paramIdx: int) : int {.inline.}  =
+  ## Sets the bindparam at the specified paramIndex to null 
+  ## (default behaviour by sqlite).
+  result = bind_null(ps, paramIdx.int32) 
+
+proc bindString*( ps: SqlPrepared, paramIdx : int,
+                   val: string,) : int =
+  ## Binds a string to the specified paramIndex.
+  result = bind_text(ps.PStmt, paramIdx.int32,val.cstring,-1.int32 , SQLITE_STATIC)
+
+proc bindBlob*( ps: SqlPrepared, paramIdx: int,val: cstring) : int  =
+  ## binds a blob to the specified paramIndex.
+  let len = val.len
+  result = bind_blob(ps.PStmt, paramIdx.int32, val[0].unSafeAddr, len.int32 , SQLITE_STATIC)
+
+proc bindParams*[T](ps: SqlPrepared, params:var T )  = 
+  var idx:int = 1
+  for _,v in params.toAny.fields:
+    case v.kind
+      of akInt:
+        if SQLITE_OK != ps.bindInt32(idx,v.getInt.int32):
+          discard
+      of akInt32:
+        if SQLITE_OK != ps.bindInt32(idx,v.getInt32):
+          discard
+      of akint64:
+        if SQLITE_OK != ps.bindInt64(idx,v.getInt64):
+          discard
+      of akFloat64:
+        if SQLITE_OK != ps.bindFloat64(idx,v.getFloat64):
+          discard
+      of akString:
+        if SQLITE_OK != ps.bindString(idx,v.getString):
+          discard
+      of akCString:
+        if SQLITE_OK != ps.bindBlob(idx,v.getCString):
+          discard
+      else:
+        discard
+    inc idx
```

Comment:
This should be a template/macro, then it can bind the type using the info available at compile time.

Using runtime information like this is dangerous.

Reply:
ok , let me try